user_workspaces_server/controllers/storagemethods/local_file_system_storage.py

```diff
@@ -7,11 +7,30 @@
 
 
 class LocalFileSystemStorage(AbstractStorage):
+    def is_valid_workspace_path(self, path):
+        # TODO: Add permission checking here
+        # The correct way to do this is to make sure that path_to_delete is a child of self.root_dir
+        # IE, path_to_delete should not be a parent of root_dir (as is the case for if path is /)
+        # it should not be a sibling of root_dir, nor should it be equal to root_dir
+        abs_root_dir = os.path.abspath(self.root_dir)
+        workspace_path = os.path.abspath(os.path.join(abs_root_dir, path))
+        if not os.path.commonpath([abs_root_dir]) == os.path.commonpath([abs_root_dir, workspace_path]):
```

Comment:
Wouldn't `os.path.commonpath([abs_root_dir])` equal `abs_root_dir`? Beyond that, using pathlib would improve readability.

Reply:
Will look at transitioning over to pathlib rather than os.path module.